### PR TITLE
Schema: Remove mixed types: road.flags (Parquet compatibility)

### PR DESCRIPTION
**Pull request by TheRBajaj**
# Description

This PR removes the mixed schema types for `road.flags` property

### Before
`road.flags` could accept multiple types of values that could be put into the same property like array and string values.

### After

`roads.flags` only accepts array property instead of array and string

# Reference

https://github.com/OvertureMaps/tf-transportation/issues/57

# Testing

Tested using test script

# Checklist

1. [x] Add relevant examples.
2. [x] Add relevant counterexamples.
3. [x] Update in-schema documentation using plain English written in complete sentences, if an update is required.
4. [x] Update Docusaurus documentation, if an update is required.
5. [ ] Review change with Overture technical writer to ensure any advanced documentation needs will be taken care of, unless the change is trivial and would not affect the documentation.

### CODE REVIEW

**Review by marinrelatic — DISMISSED** [review]

**Review by brad-richardson — DISMISSED** [review]

**Review by bastiaanvanassche-tomtom — APPROVED** [review]

**Review by sanjibdutta1 — APPROVED** [review]

**Review by vcschapp — CHANGES_REQUESTED** [review]
@TheRBajaj need to fix counterexamples.

1. [ ] There are many existing `counterexamples` that have the old-style flags, but where the intention of those counterexamples isn't to find an error in the `road.flags` property. Can you please find and fix them all to use new-style flags?
2. [x] Please get in the habit of looking for counterexamples to fix when doing schema changes!
3. [ ] Could you also add one counterexample that explicitly tests for invalid old-style flags?

**Comment by vcschapp** [comment]
@TheRBajaj are you using the pull request template?

Looking at this PR's description, it looks like the template was deleted or trimmed-down. If you're doing that, it's preferable not to.

These are the benefits of the PR template:

1. It has a standard format, so it makes it easier to find expected information in expected places.
2. It helps you not forget things.
3. It has a checklist. Explicitly on the checklist, there's an entry for "Add relevant counterexamples", which would help you remember to do that in this case.

**Review by vcschapp — APPROVED** [review]
```
                   __
                  / \--..____
                   \ \       \-----,,,..
                    \ \       \         \--,,..
                     \ \       \         \  ,'
                      \ \       \         \ ``..
                       \ \       \         \-''
                        \ \       \__,,--'''
                         \ \       \.
                          \ \      ,/
                           \ \__..-
                            \ \
                             \ \
                              \ \   :F_P:
                               \ \
                                \ \
                                 \ \
                                  \ \
                                   \ \
                                    \ \
```

File: counterexamples/transportation/segment/road/restrictions/turns/bad-turns-reason.yaml
Comment (brad-richardson) [inline counterexamples/transportation/segment/road/restrictions/turns/bad-turns-reason.yaml:23]: Shouldn't these be `is_private`?

**Review by brad-richardson — CHANGES_REQUESTED** [review]

**Review by brad-richardson — DISMISSED** [review]
It looks like you have some conflicts to resolve first but LGTM!

**Review by sanjibdutta1 — APPROVED** [review]

**Review by ibnt1 — APPROVED** [review]

**Comment by vcschapp** [comment]
> It looks like you have some conflicts to resolve first but LGTM!

I see it also. @TheRBajaj could you resolve the conflict to make this PR mergable?

**Comment by vcschapp** [comment]
Merging this despite the fact that validation is currently broken. Some cross-PR conflicts caused merging several previous PRs that by themselves were fine to break validation. It will be easier to merge this first, then fix after.

**Pull request merged**
